test/test_autograd.py

```diff
@@ -7016,6 +7026,249 @@ def test_size_check(self):
 
             dual = fwAD.make_dual(foo, tangent[1:])
 
+    # The following test functions want to ensure all the following behaviors:
+    #   - Ensure that default level system in the python binding works
+    #   - Ensure that only level 0 exists and nesting is properly disabled
+    #   - Ensure that printing works fine
+    #   - Ensure that basic packing/unpacking works
+    #   - Ensure that advanced packing/unpacking works
+    #     - For memory / version counter share
+    #     - For backward AD (regular ops)
+    #   - Ensure that view + inplace for both modes work fine
+    #   - Ensure we do proper cleanup on exit of a level
+
+    def test_default_level(self):
+        foo = torch.rand(2)
+        bar = torch.rand(2)
+
+        with fwAD.dual_level():
+            baz = fwAD.make_dual(foo, bar)
```

Comment:
Is there a test for what happens when:
1. `foo` and `bar` have different sizes?
2. `foo` and `bar` have the same sizes but different strides?
3. `foo` and `bar` have different dtypes?
4. `foo` and `bar` have different devices?